Source/Basic-Conditions-And-Behaviors/Runtime/Conditions/CollectedCondition.cs

```diff
@@ -0,0 +1,141 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.Runtime.Serialization;
+using VRBuilder.Core.Attributes;
+using VRBuilder.Core.SceneObjects;
+using UnityEngine;
+using VRBuilder.Core.Properties;
+
+namespace VRBuilder.Core.Conditions
+{
+    /// <summary>
+    /// A condition that checks if new selected objects are entering the selected collider.
+    /// </summary>
+    [DataContract(IsReference = true)]
+    public class CollectedCondition: Condition<CollectedCondition.EntityData>
+    {
+        [DataContract(IsReference = true)]
+        [DisplayName("Collect Condition")]
+        public class EntityData: IConditionData
+        {
+            [HideInProcessInspector]
+            public readonly List<GameObject> Collected = new();
+
+            /// <summary>
+            /// The objects that has to enter the collider.
+            /// </summary>
+            [DataMember]
+            [DisplayName("Object")]
+            public MultipleSceneObjectReference TargetObjects { get; set; }
+
+            /// <summary>
+            /// The collider with trigger to enter.
+            /// </summary>
+            [DataMember]
+            [DisplayName("Collider")]
+            public SingleScenePropertyReference<ColliderWithTriggerProperty> TriggerObject { get; set; }
+
+            public Metadata Metadata { get; set; }
+
+            /// <inheritdoc />
+            [HideInProcessInspector]
+            [IgnoreDataMember]
+            public string Name => $"Collect unique {TargetObjects} in {TriggerObject}";
```

Comment:
```suggestion
            public string Name => $"Place one of {TargetObjects} in {TriggerObject}";
```
TargetObjects to string will either return something like "objects in MyGroup" or "15 objects" (or a single object's name, but we can expect people to reference multiples here). I would rephrase to ensure the sentence makes sense.